### PR TITLE
Lint ./docs

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

* Un-ignore ./docs for linting
* Fix `./docs/*.md` lint errors
* Moved override for `*.md` files after the override for `*.ts` files to avoid overwriting some rules.

#### Testing instructions

N/A

### CODE REVIEW

**Review by tyxla — APPROVED** [review]
Looks great to me, thanks! 🚢

File: docs/testing/unit-tests.md [hunk 1]
```diff
@@ -1,36 +1,42 @@
...
-    * [Importing tests](#importing-tests)
-    * [Describing tests](#describing-tests)
-    * [Snapshot testing](#snapshot-testing)
-    * [Setup and Teardown methods](#setup-and-teardown-methods)
-    * [Mocking dependencies](#mocking-dependencies)
-    * [Testing globals](#testing-globals)
-    * [Testing legacy code](#testing-legacy-code)
-* [Further reading](#further-reading)
+
+- [Unit Tests](#unit-tests)
```
Comment (tyxla) [inline docs/testing/unit-tests.md:7]: Wanted to note that this change felt a bit odd the first glance, but it makes a lot of sense after previewing the parsed document 👌

File: docs/coding-guidelines/typescript.md [hunk 2]
```diff
@@ -85,13 +86,15 @@ type of data and supplying _what you want_.
 Because TypeScript infers most types don't feel obligated to add types to everthing, especially literal values.
 
 #### Avoid
+
 ```ts
-const name: string = 'Calypso';
-const isDesktopBuild: boolean = false;
```
Comment (dmsnell) [inline docs/coding-guidelines/typescript.md:90]: @scinos this change kind of defeats the purpose of the "avoid" example 🙃

Comment (dmsnell) [reply docs/coding-guidelines/typescript.md:90]: also, why on earth would a linter mess with the contents inside a fenced code block?

Comment (scinos) [reply docs/coding-guidelines/typescript.md:90]: Linter can be disabled with the usual `//eslint-disable` and friends. I missed this one :(

The main goal of linting code blocks is to make sure our docs follow our standard, so anybody can copy/paste a piece of code form our doc and it is ready to go. I agree it gets a bit muddy when we have examples of what _not_ to do.

Comment (dmsnell) [reply docs/coding-guidelines/typescript.md:90]: do you want me to open a PR to undo the lint breakage or do you have that?

Comment (scinos) [reply docs/coding-guidelines/typescript.md:90]: I'll fix it, don't worry. Thanks for the offer anyway :)

Comment (dmsnell) [reply docs/coding-guidelines/typescript.md:90]: thanks!

Comment (scinos) [reply docs/coding-guidelines/typescript.md:90]: #49584

**Pull request merged**